### PR TITLE
Implement Views for Users and Posts

### PR DESCRIPTION
**Pull request by cancelei**
Hello Reviewer,

In this PR I implemented a lot of code and updates to accommodate the requirements of the project milestone.

**Highlights:**

- Implement the design from the sneak peek wireframes.

Aditional improvements:
Update of Seed Data Base with 15 users that each have 10 posts and 15 comments each, which makes the app load "slower" when loading it. Keep that in mind when testing.

Good review!

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
+      <div class="user-bio">
+        <h3>Bio</h3>
+        <p><%= @user.bio %></p>
+      </div>
+    </div>
+  </div>
+
+  <div class="user-posts">
+    <h3>Posts:</h3>
+    <% @user.posts.each do |post| %>
```
Comment (Davidosky007) [inline app/views/users/show.html.erb]: * According to the wireframe you are to have a method that returns 3 most recent posts of the user so in your user show view file you will render the posts like this `<% @user.recent_post.each do |post| %>` 
image

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1,6 +1,33 @@
...
-    <li><%= link_to post.title, user_post_path(@user, post) %></li>
+    <div class="post-card">
+      <h3><%= post.title %></h3>
+      <p><%= post.body.truncate(100) %></p>
+      <div class="post-stats">
+        <span>Comments: <%= post.comments_counter %></span>
+        <span>Likes: <%= post.likes_counter %></span>
+      </div>
+      <div class="post-comments">
+        <% post.comments.each do |comment| %>
```
Comment (Davidosky007) [inline app/views/posts/index.html.erb]: * Also here According to the wireframe you are to have a method that returns 5 most recent posts of the user so in your post index view file you will render the comments like this ``<% post.recent_comments.each do |comment| %>`` 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @cancelei 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* All views setup properly 👍 
* Well-written PR 👍 
* No linter errors 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1,5 +1,32 @@
...
+    <% @user.posts.take(3).each do |post| %>
+      <div class="user-post">
+        <h4><%= link_to post.title, user_post_path(@user, post) %></h4>
+        <p><%= post.body.truncate(100) %></p>
+        <div class="post-stats">
+          <span>Comments: <%= post.comments_counter %></span>
+          <span>Likes: <%= post.likes_counter %></span>
+        </div>
+      </div>
+    <% end %>
```
Comment (shella12) [inline app/views/users/show.html.erb]: - [x] As mentioned by the previous reviewer when we are visiting `users/:id` page there should be 3 most recent posts by the user on display and you can easily do that by making use of the method in your user model `recent_posts`. Example snippet:

```rb

# app/views/users/show.html.erb

[...]

<% @user.recent_post.each do |post| %>

<h4><%= link_to post.title, user_post_path(post) %></h4>
<p><%= post.text %></p>
<p>
<small>Comments: <%= post.comments_counter || 0 %>,</small> 
<small>Likes: <%= post.likes_counter || 0 %></small>
</p>

<% end %>

<button>
<%= link_to 'See all Post', user_posts_path(@user) %>
</button>


```

WireFrame| your App
--|--
Screenshot 2023-10-13 at 7 29 37 PM|Screenshot 2023-10-13 at 7 19 23 PM

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1,6 +1,33 @@
...
+      <div class="post-stats">
+        <span>Comments: <%= post.comments_counter %></span>
+        <span>Likes: <%= post.likes_counter %></span>
+      </div>
+      <div class="post-comments">
+        <% post.comments.take(5).each do |comment| %>
+          <div class="comment">
+            <strong><%= comment.user.name %>:</strong> <%= comment.content %>
+          </div>
+        <% end %>
```
Comment (shella12) [inline app/views/posts/index.html.erb]: - [x] As mentioned by the previous reviewer please show the five most recent comments under the relevant post when we are on `users/:id/posts` page and you can easily do that by making use of the methods in your post model `recent_comments`. Example snippet: 

```rb
 <% post.recent_comments.each do |comment| %>
  <p><%= comment.author.name %>: <%= comment.text %></p> 
 <% end %>

```

WireFrame| your App
--|--
Screenshot 2023-10-13 at 7 29 50 PM|Screenshot 2023-10-13 at 7 30 46 PM

**Review by shella12 — CHANGES_REQUESTED** [review]
Hi @cancelei, 

Great work on making the changes requested by a previous reviewer 👏 👏 👏 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet. You are almost there...



## Highlights

- Added required view files ✅ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @cancelei, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [x] All requested changes are made ✔️ 
- [x] Correct Gitflow ✔️ 
- [x] No linters errors ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**